Execute Procedure Expect Exception in standalone mode

MMatten opened the pull request:
Adds support for the `Execute Procedure Expect Exception` in standalone mode.

javornikolov commented on FitNesseRoot/DbFit/AcceptanceTests/JavaTests/OracleTests/LibraryTests/StoredProcWithException/content.txt:
```diff
@@ -1,12 +1,12 @@
-          raise_application_error(-20001,'test exception');
-        end if;
-	strlength:=length(name);
+    if (name='xx') then
```
If we're touching this, might be good to refine a few other things: add space around `=`, `:=`, and after `,`.

MMatten replied:
Now applied.

Perhaps we should adopt some standard SQL formatting via some free web service such as http://www.dpriver.com/pp/sqlformat.htm, but I don't know how we could define a standard config of our chosing. Perhaps use the service's default. Or maybe there are other better free desktop tools.

javornikolov commented on dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java:
```diff
@@ -0,0 +1,39 @@
+    @Override
+    protected ExpectedBehaviour getExpectedBehaviour() {
+        if (!excNumberDefined) return ExpectedBehaviour.ANY_EXCEPTION;
+        return ExpectedBehaviour.SPECIFIC_EXCEPTION;
```
May we simplify that a bit? It should be possible to fit it in a single 1-line operator.

MMatten replied:
Updated.

MMatten commented:
I've added some missing content/properties (now .wiki) files too for the new `LibraryTests` root and fixed a couple of issues with the symlinks.

javornikolov commented on dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java:
```diff
@@ -0,0 +1,38 @@
+
+    @Override
+    protected ExpectedBehaviour getExpectedBehaviour() {
+        return excNumberDefined ? ExpectedBehaviour.SPECIFIC_EXCEPTION : ExpectedBehaviour.ANY_EXCEPTION;
```
Maybe would look better if we can skip the prefix `ExpectedBehaviour.` here?

MMatten replied:
Agreed. What do you think of it now?

Is it better to only import the minimal number of references?

javornikolov replied:
Yes, it looks good now. What do you mean by min number of references? (In general I think we have a Checkstyle will raise an error if we have unused imports).

javornikolov commented:
> I've added some missing content/properties (now .wiki) files too for the new LibraryTests root and fixed a couple of issues with the symlinks.

What do you think about extracting this refactoring and small fixes to a separate PR? This would keep the current PR focused on the Execute Procedure Expect Exception change.

MMatten commented:
These changes are directly related to the changes to the tests (creating the `LibraryTests` container). They weren't existing issues.

Sent from my iPhone

On 18 Mar 2017, at 13:15, Yavor Nikolov <notifications@github.com<mailto:notifications@github.com>> wrote:


I've added some missing content/properties (now .wiki) files too for the new LibraryTests root and fixed a couple of issues with the symlinks.

What do you think about extracting this refactoring and small fixes to a separate PR? This would keep the current PR focused on the Execute Procedure Expect Exception change.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/dbfit/dbfit/pull/556#issuecomment-287545736>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABztFQVQoCZEDvIsfk5GX3xeXt6Spb-oks5rm9kJgaJpZM4MhMRv>.

javornikolov commented:
> These changes are directly related to the changes to the tests (creating the `LibraryTests` container). They weren't existing issues.

Ah, OK then.

MMatten commented:
I meant the use of import with `.*` instead of specific items.


On 18 Mar 2017, at 13:46, Yavor Nikolov <notifications@github.com<mailto:notifications@github.com>> wrote:


@javornikolov commented on this pull request.

________________________________

In dbfit-java/core/src/main/java/dbfit/fixture/ExecuteProcedureExpectException.java<https://github.com/dbfit/dbfit/pull/556#discussion_r106781225>:

> +                            int expectedErrorCode) {
+        this.procName = procName;
+        this.environment = dbEnvironment;
+        this.excNumberDefined = true;
+        this.excNumberExpected = expectedErrorCode;
+    }
+
+    public ExecuteProcedureExpectException(DBEnvironment dbEnvironment, String procName) {
+        this.procName = procName;
+        this.environment = dbEnvironment;
+        this.excNumberDefined = false;
+    }
+
+    @Override
+    protected ExpectedBehaviour getExpectedBehaviour() {
+        return excNumberDefined ? ExpectedBehaviour.SPECIFIC_EXCEPTION : ExpectedBehaviour.ANY_EXCEPTION;


Yes, it looks good now. What do you mean by min number of references? (In general I think we have a Checkstyle will raise an error if we have unused imports).

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/dbfit/dbfit/pull/556#discussion_r106781225>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABztFYXpiNUKUr7PfcNLiHc_zBnTmAnCks5rm-AngaJpZM4MhMRv>.

javornikolov commented:
> I meant the use of import with `.*` instead of specific items.

It depends. In general it's not very good to "pollute" the namespace with too many things (there might be naming conflicts). In case of enums in particular (which is the case here) - I guess it would be usually OK to do a static * import.

javornikolov commented:
The current state seems OK for this PR to be merged after some squashing of the fix-up commits.

MMatten commented:
> The current state seems OK for this PR to be merged after some squashing of the fix-up commits.

Ooops. Squashed.

javornikolov commented:
All looks great now :+1: Merging... Thank you @MMatten!
